File: cgi-bin/form_post.php
<?php
if ($stdin = fopen('php://input', 'r')) {
    // Read all lines from stdin
    $queryString = '';
    while ($line = fgets($stdin)) {
        $queryString .= $line;
    }
    fclose($stdin);
} else {
    // If stdin is not available, set an error message
    $queryString = "No input received from stdin";
}

$body = "<!DOCTYPE html>
<html>
<head>
<title>Received POST Data</title>
</head>
<body>
<h1>Info passed by stdin was:</h1>
<p>$queryString</p>
</body>
</html>";

// Calculate content length
$length = strlen($body);

// Set HTTP headers
header("Content-type: text/html; charset=UTF-8");

// Output the body
echo $body;
?>

File: cgi-bin/form_post.py
#!/usr/bin/python3
import os
import sys
import urllib.parse

input_string = sys.stdin.read()
print(input_string)
print ("Content-type: text/html; charset=UTF-8")
print()
print ('''<!DOCTYPE html>
<html>
<head>
<title>Received POST Data</title>
</head>
<body>
<h1>Info passed by stdin was:</h1>
<p> ''')
print (input_string)
print ('''</p>
<p>AFTER DECODING: </p>''')
print (urllib.parse.parse_qs(input_string))    
print('''
</body>
</html>''')
